plugins/slo/pkg/slo/interfaces.go

```diff
@@ -0,0 +1,64 @@
+package slo
+
+import (
+	"context"
+
+	v1 "github.com/alexandreLamarre/oslo/pkg/manifest/v1"
+	"github.com/hashicorp/go-hclog"
+	"github.com/rancher/opni/pkg/slo/shared"
+	apis "github.com/rancher/opni/plugins/slo/pkg/apis/slo"
+	sloapi "github.com/rancher/opni/plugins/slo/pkg/apis/slo"
+	"google.golang.org/protobuf/proto"
+)
+
+var datasourceToImpl map[string]SLOImpl = make(map[string]SLOImpl)
+
+func init() {
+	datasourceToImpl[shared.MonitoringDatasource] = SLOMonitoring{}
+}
+
+type SLOImpl interface {
```

Comment:
the naming here is a bit weird. I'd avoid the name `impl` for interfaces or interface methods. A better name for this interface might be `Backend` or `Store` (such that the full name is `slo.Backend` or `slo.Store` which reads better in the context of the package

Reply:
You're right, I think here my idea was because I was using the message `SLOImplData` as the data stored in the backend. Something like : "This is all the data that an SLO implementation needs". 

I think I should also rename the message to match the new interface name

Reply:
Renamed the interfaces based on the suggested naming conventions, still need to rename the `SloImplData` Message

Reply:
Done